src/lib/services/segment-service.ts

```diff
@@ -90,39 +93,50 @@ export class SegmentService implements ISegmentService {
     async getVisibleStrategies(
         id: number,
         userId: number,
-    ): Promise<IFeatureStrategy[]> {
-        const strategies = await this.getAllStrategies(id);
+    ): Promise<StrategiesUsingSegment> {
+        const allStrategies = await this.getAllStrategies(id);
         if (this.flagResolver.isEnabled('privateProjects')) {
             const accessibleProjects =
                 await this.privateProjectChecker.getUserAccessibleProjects(
                     userId,
                 );
             if (accessibleProjects.mode === 'all') {
-                return strategies;
+                return allStrategies;
             } else {
-                return strategies.filter((strategy) =>
-                    accessibleProjects.projects.includes(strategy.projectId),
-                );
+                const filter = (strategy) =>
+                    accessibleProjects.projects.includes(strategy.projectId);
+                return {
+                    strategies: allStrategies.strategies.filter(filter),
+                    changeRequestStrategies:
+                        allStrategies.changeRequestStrategies.filter(filter),
+                };
             }
         }
-        return strategies;
+        return allStrategies;
     }
 
-    async getAllStrategies(id: number): Promise<IFeatureStrategy[]> {
+    async getAllStrategies(id: number): Promise<StrategiesUsingSegment> {
         const strategies =
             await this.featureStrategiesStore.getStrategiesBySegment(id);
-        return strategies;
+
+        const strategyIds = new Set(strategies.map((s) => s.id));
+
+        const changeRequestStrategies = (
+            await this.changeRequestSegmentUsageReadModel.getStrategiesUsedInActiveChangeRequests(
+                id,
+            )
+        ).filter(
+            (strategy) => !('id' in strategy && strategyIds.has(strategy.id)),
+        );
```

Comment:
This is the point of contention mentioned in the PR description: should we filter out duplicate IDs or not?